code/modules/projectiles/guns/energy.dm

```diff
@@ -62,36 +64,42 @@
 		if(!power_supply)
 			return
 		power_supply.give(100)
+		if(!chambered) //if empty chamber we try to charge a new shot
+			recharge_newshot(1)
 		update_icon()
 
 /obj/item/weapon/gun/energy/attack_self(mob/living/user as mob)
 	if(ammo_type.len > 1)
 		select_fire(user)
 		update_icon()
 
-/obj/item/weapon/gun/energy/afterattack(atom/target as mob|obj|turf, mob/living/user as mob|obj, params)
-	newshot() //prepare a new shot
-	..()
-
 /obj/item/weapon/gun/energy/can_shoot()
 	var/obj/item/ammo_casing/energy/shot = ammo_type[select]
 	return power_supply.charge >= shot.e_cost
 
-/obj/item/weapon/gun/energy/newshot()
+/obj/item/weapon/gun/energy/recharge_newshot(no_cyborg_drain)

```

Comment:
the argument is here to prevent cyborgs from losing power when selecting a different ammo type for example (if we ever gave the HoS gun to a borg).